apps/src/sites/studio/pages/signup.js

```diff
@@ -168,30 +168,45 @@ window.SignupManager = function (options) {
     logAnalyticsEvent(event);
   }
 
+  function logEventWithInferredUserType(event, extraData = {}) {
+    if (isTeacherSelected()) {
+      logAnalyticsEvent(event + "_teacher", extraData);
+    } else if (isStudentSelected()) {
+      logAnalyticsEvent(event + "_student", extraData);
+    } else {
+      logAnalyticsEvent(event + "_no_user_type", extraData);
+    }
+  }
+
   function logFormSubmitted() {
-    const event = isTeacherSelected() ? "submit_teacher" : "submit_student";
-    logAnalyticsEvent(event);
+    logEventWithInferredUserType("submit");
   }
 
   function logFormError(err) {
-    const event = isTeacherSelected() ? "submit_error_teacher" : "submit_error_student";
-    logAnalyticsEvent(event, {error_info: err});
+    logEventWithInferredUserType("submit_error", {error_info: err});
   }
 
   function logFormSuccess() {
-    const event = isTeacherSelected() ? "submit_success_teacher" : "submit_success_student";
-    logAnalyticsEvent(event);
+    logEventWithInferredUserType("submit_success");
   }
 
-  function isTeacherSelected() {
+  function isUserTypeSelected(user_type) {
     const formData = $('#new_user').serializeArray();
     const userType = $.grep(formData, e => e.name === "user[user_type]");
-    if (userType.length === 1 && userType[0].value === "teacher") {
+    if (userType.length === 1 && userType[0].value === user_type) {
       return true;
     }
     return false;
   }
```

Comment:
As written, I need to turtle pretty far down into the method chain to bottom out, making it hard to understand. It seems cleaner to write
```
function getUserTypeSelected() {
  const formData = $('#new_user').serializeArray();
  const userType = $.grep(formData, e => e.name === "user[user_type]");
  if (userType.length === 1) {
    return userType[0].value;
  }
  return "no_user_type";
}
```
and
```
function logEventWithInferredUserType(event, extraData = {}) {
  userTypeSuffix = getUserTypeSelected();
  logAnalyticsEvent(event + userTypeSuffix, extraData);
}
```